src/conversion.rs

```diff
@@ -502,6 +587,68 @@ impl IntoPy<Py<PyTuple>> for () {
     fn into_py(self, py: Python<'_>) -> Py<PyTuple> {
         PyTuple::empty(py).unbind()
     }
+
+    #[inline]
+    fn __py_call_vectorcall1<'py>(
+        self,
+        py: Python<'py>,
+        function: Borrowed<'_, 'py, PyAny>,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        unsafe {
+            cfg_if::cfg_if! {
+                if #[cfg(all(
+                    not(PyPy),
+                    not(GraalPy),
+                    any(Py_3_10, all(not(Py_LIMITED_API), Py_3_9)) // PyObject_CallNoArgs was added to python in 3.9 but to limited API in 3.10
+                ))] {
+                    // Optimized path on python 3.9+
+                    ffi::PyObject_CallNoArgs(function.as_ptr()).assume_owned_or_err(py)
+                } else {
+                    ffi::PyObject_Call(function.as_ptr(), PyTuple::empty(py).as_ptr(), std::ptr::null_mut()).assume_owned_or_err(py)
+                }
+            }
```

Comment:
We have just recently started defining "compat" functions in `pyo3-ffi/compat`. I wonder if this makes sense to move there already.

Reply:
If I could define `PyObject_CallNoArgsCompat()` too, maybe, but since I can't (see the other comment) I don't believe this is a good idea. There is no code duplication, and the code is pretty clear (aside from the ugly `<Self as IntoPy<Py<PyTuple>>>` because of inference problems).

Reply:
Please rebase and use the symbol I now added in #4461 😄